autoload/lsp/completion.vim

```diff
@@ -311,7 +312,9 @@ def g:LspOmniFunc(findstart: number, base: string): any
       return res->empty() ? v:none : res
     endif
 
-    return res->empty() ? v:none : res->filter((i, v) => v.word =~# '^' .. lspserver.omniCompleteKeyword)
+    var prefix: string = lspserver.omniCompleteKeyword->tolower()
+    # To filter (case ignored) keyword prefixed compl items only.
+    return res->empty() ? v:none : res->filter((i, v) => v.word->tolower()->stridx(prefix) == 0)
```

Comment:
What are you trying to archive? Allow incase-sensitive completion? If so please  create an option for it. Possible one that can also take `fuzzy`.

Some language servers filters the completion items, see `clangd` for an example on this, other relays on the client doing the filtering, see `typescript-language-server` for such a language server.

This code will only have affect for the later case, keep that in mind.

If an option is added, then I suggest that the line 132 above:

```
if prefix != '' && d.word->tolower()->stridx(prefix) != 0
```

is changed to respect this option as well.